### PR TITLE
feat: Support environments without ws by dynamically importing WebSocket module with error handling

### PR DESCRIPTION
**Pull request by tomkosm**

### CODE REVIEW

File: apps/js-sdk/firecrawl/src/index.ts [hunk 1]
```diff
@@ -1,7 +1,24 @@
...
+ * If the import fails, WebSocket is set to null.
+ * This approach is used because some environments, such as Firebase Functions,
+ * might not support WebSocket natively.
+ */
+const WebSocket: typeof IsowsWebSocket | null = await (async () => {
+  try {
+    const module = await import('isows');
+    return module.WebSocket;
+  } catch (error) {
+    console.error("Failed to load 'isows' module:", error);
```
Comment (mogery) [inline apps/js-sdk/firecrawl/src/index.ts]: Let's not log this.

**Review by mogery — CHANGES_REQUESTED** [review]

**Review by mogery — APPROVED** [review]

**Comment by tomkosm** [comment]
> blocked by fc side, will merge then

Why is it blocked? Dont merge yet need to do some more testing

**Comment by tomkosm** [comment]
@mogery @rafaelsideguide should be good to go, added testing and fixed some tests that were broken

**Comment by nickscamara** [comment]
```CJS Build failed
Error: Build failed with 1 error:
src/index.ts:12:48: ERROR: Top-level await is currently not supported with the "cjs" output format
    at addChunk (node:internal/streams/readable:368:12)```

**Pull request merged**